src/Sylius/Bundle/ThemeBundle/Context/ThemeContext.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Sylius\Bundle\ThemeBundle\Context;
+
+use Sylius\Bundle\ThemeBundle\Model\ThemeInterface;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+class ThemeContext implements ThemeContextInterface
+{
+    /**
+     * @var ThemeInterface[]
+     */
+    private $themes = [];
+
+    /**
+     * Keys are logical names of themes, values are priorities.
+     *
+     * @var array
+     */
+    private $themesPriorities = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getThemes()
+    {
+        return $this->themes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTheme($logicalName)
+    {
+        foreach ($this->themes as $theme) {
+            if ($logicalName === $theme->getLogicalName()) {
+                return $theme;
+            }
+        }
+
+        return null;
```

Comment:
Why can't we keep Themes in array with logical name as an array key? We can then do the following:

```php
return $this->themes[$logicalName];
```